symExec.py

```diff
@@ -897,31 +890,25 @@ def sym_exec_ins(start, instr, stack, mem, global_state, path_conditions_and_var
             first = stack.pop(0)
             second = stack.pop(0)
             third = stack.pop(0)
-            if isinstance(third, (int, long)):
+
+            if contains_only_concrete_values([first, second, third]):
                 if third == 0:
                     computed = 0
                 else:
-                    if not (isinstance(first, (int, long)) and isinstance(second, (int, long))):
-                        # there is one guy that is a symbolic expression
-                        third = BitVecVal(third, 256)
-                        if isinstance(first, (int, long)):
-                            first = BitVecVal(first, 256)
-                        if isinstance(second, (int, long)):
-                            second = BitVecVal(second, 256)
                     computed = (first * second) % third
             else:
-                solver.push()
-                solver.add(Not(third == 0))
-                if solver.check() == unsat:
-                    # it is provable that second is indeed equal to zero
+                first = to_symbolic(first)
+                second = to_symbolic(second)
+                solver.add(third == 0)
+                if solver.check() == sat:
                     computed = 0
                 else:
-                    if isinstance(first, (int, long)):
-                        first = BitVecVal(first, 256)
-                    if isinstance(second, (int, long)):
-                        second = BitVecVal(second, 256)
-                    computed = (first * second) % third
-                solver.pop()
+                    first = ZeroExt(256, first)
```

Comment:
Hey ZeroExt seems to add 256 _extra_ zero bits. We want the total size to be 256 bits right?